File: LOO 2°Parcial/PY's/EmpleadoPiezas.py
# -*- coding: utf-8 -*-
"""
Created on Fri Mar 19 12:23:24 2021

@author: MedNo
"""

import Empleado as Emp

class EmpleadoPiezas(Emp.Empleado):
    def __init__(self, nombre="", cantidad=0, pago=0):
        Emp.Empleado.__init__(self,nombre)
        if cantidad>0:
            self.__cantidadpiezas = cantidad
        else:
            self.__cantidadpiezas = 0
        if pago > 0:
            self.__pagoxpieza = pago
        else:
            self.__pagoxpieza = 0
    
    @property 
    def cantidadpiezas(self):
        return self.__cantidadpiezas
    @cantidadpiezas.setter
    def cantidadpiezas(self, cantidad):
        if cantidad>0:
            self.__cantidadpiezas = cantidad
        else:
            self.__cantidadpiezas = 0
    @property
    def pagoxpieza(self):
        return self.__pagoxpieza
    @pagoxpieza.setter
    def pagoxpieza(self, pago):
        if pago > 0:
            self.__pagoxpieza = pago
        else:
            self.__pagoxpieza = 0
    
    def pagoDiario(self):
        return self.__cantidadpiezas * self.__pagoxpieza
    
    def imprimir(self):
        print("Empleado tipo TRABAJADOR POR PIEZAS");
        print("NOMBRE: " + self.nombre + "\nPAGO POR PIEZA: " + str(self.__pagoxpieza) + "\nCANTIDAD PIEZAS HECHAS: " + str(self.__cantidadpiezas) + "\nPAGO DIARIO: " + str(self.pagoDiario()))
    

File: LOO 2°Parcial/EmpleadosCSharp/EmpleadosCSharp/EmpleadoPiezas.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EmpleadosCSharp
{
    class EmpleadoPiezas : Empleado
    {
        protected int cantidadPiezas;
        protected double pagoXPieza;
        public EmpleadoPiezas() : base()
        {
            this.cantidadPiezas = 0;
            this.pagoXPieza = 0;
        }
        public EmpleadoPiezas(String nombre, int cantidad, double pago) : base(nombre)
        {
            this.cantidadPiezas = cantidad > 0 ? cantidad : 0;
            this.pagoXPieza = pago > 0 ? pago : 0;
        }
        public int CantidadPiezas
        {
            set { this.cantidadPiezas = value > 0 ? value : 0; }
            get { return this.cantidadPiezas; }
        }
        public double PagoXPieza
        {
            set { this.pagoXPieza = value > 0 ? value : 0; }
            get { return this.pagoXPieza; }
        }
        public override double pagoDiario()
        {
            return this.cantidadPiezas * this.pagoXPieza;
        }
        public override void imprimir()
        {
            Console.WriteLine("Empleado tipo TRABAJADOR POR PIEZAS");
            Console.WriteLine("NOMBRE: " + this.nombre + "\nPAGO POR PIEZA: " + this.pagoXPieza + "\nCANTIDAD PIEZAS HECHAS: " + this.cantidadPiezas + "\nPAGO DIARIO: " + this.pagoDiario());
        }
    }
}